e2e/tests-dfx/assetscanister.bash

```diff
@@ -54,16 +54,29 @@ teardown() {
   assert_command dfx canister call e2e_project_frontend list_authorized '()'
   assert_contains "$STRANGER_PRINCIPAL"
 
-  # authorized principals, that are not controllers, can authorize other principals
+  # authorized principals, that are not controllers, cannot authorize other principals
   assert_command dfx canister call e2e_project_frontend authorize "(principal \"$AUTHORIZED_PRINCIPAL\")" --identity controller
-  assert_command dfx canister call e2e_project_frontend authorize "(principal \"$BACKDOOR_PRINCIPAL\")" --identity authorized
+  assert_command_fail dfx canister call e2e_project_frontend authorize "(principal \"$BACKDOOR_PRINCIPAL\")" --identity authorized
+
+  # Why are these different? https://dfinity.atlassian.net/browse/SDK-955 will find out.
+  [ "$USE_IC_REF" ] && assert_contains "canister did not respond"
+  [ ! "$USE_IC_REF" ] && assert_contains "Caller is not a controller"
+
+  assert_command dfx canister call e2e_project_frontend list_authorized '()'
+  assert_not_contains "$BACKDOOR_PRINCIPAL"
+
+  # authorized principals, that are not controllers, cannot deauthorize others
+  assert_command dfx canister call e2e_project_frontend authorize "(principal \"$BACKDOOR_PRINCIPAL\")" --identity controller
   assert_command dfx canister call e2e_project_frontend list_authorized '()'
   assert_contains "$BACKDOOR_PRINCIPAL"
+  assert_command_fail dfx canister call e2e_project_frontend deauthorize "(principal \"$BACKDOOR_PRINCIPAL\")" --identity authorized
+
+  # Why are these different? https://dfinity.atlassian.net/browse/SDK-955 will find out.
+  [ "$USE_IC_REF" ] && assert_contains "canister did not respond"
+  [ ! "$USE_IC_REF" ] && assert_contains "Caller is not a controller"
 
-  # authorized principals, that are not controllers, can deauthorize others
-  assert_command dfx canister call e2e_project_frontend deauthorize "(principal \"$BACKDOOR_PRINCIPAL\")" --identity authorized
   assert_command dfx canister call e2e_project_frontend list_authorized '()'
-  assert_not_contains "$BACKDOOR_PRINCIPAL"
+  assert_contains "$BACKDOOR_PRINCIPAL"
```

Comment:
Still calling it `BACKDOOR_PRINCIPAL` sounds wrong to me - it sounds like it checks that it's still possible to sneak in principals that shouldn't be there

Reply:
Good point.  I will follow up.